### PR TITLE
Added ability to use     PDO_MYSQL_ATTR_SSL_SERVER_CN to set the serv…

### PR DESCRIPTION
**Pull request by jerairrest**
…er name when using pdo ssl

### CODE REVIEW

File: ext/pdo_mysql/pdo_mysql.c [hunk 1]
```diff
@@ -130,6 +130,7 @@ static PHP_MINIT_FUNCTION(pdo_mysql)
 	 REGISTER_PDO_CLASS_CONST_LONG("MYSQL_ATTR_SERVER_PUBLIC_KEY", (zend_long)PDO_MYSQL_ATTR_SERVER_PUBLIC_KEY);
 #endif
 	REGISTER_PDO_CLASS_CONST_LONG("MYSQL_ATTR_MULTI_STATEMENTS", (zend_long)PDO_MYSQL_ATTR_MULTI_STATEMENTS);
+	REGISTER_PDO_CLASS_CONST_LONG("MYSQL_ATTR_SSL_SERVER_CN", (long)PDO_MYSQL_ATTR_SSL_SERVER_CN);
```
Comment (rickywiens) [inline ext/pdo_mysql/pdo_mysql.c]: zend_long here maybe?

File: ext/pdo_mysql/mysql_driver.c [hunk 2]
```diff
@@ -721,6 +723,9 @@ static int pdo_mysql_handle_factory(pdo_dbh_t *dbh, zval *driver_options)
 			if (ssl_cipher) {
 				zend_string_release(ssl_cipher);
 			}
+			if(ssl_server_cn){
```
Comment (rickywiens) [inline ext/pdo_mysql/mysql_driver.c]: indentation here

File: ext/mysqlnd/mysqlnd.h [hunk 3]
```diff
@@ -158,7 +158,7 @@ PHPAPI enum_func_status mysqlnd_poll(MYSQLND **r_array, MYSQLND **e_array, MYSQL
 PHPAPI const char *	mysqlnd_get_client_info();
 PHPAPI unsigned int	mysqlnd_get_client_version();
 
-#define mysqlnd_ssl_set(conn, key, cert, ca, capath, cipher) ((conn)->data)->m->ssl_set((conn)->data, (key), (cert), (ca), (capath), (cipher))
+#define mysqlnd_ssl_set(conn, key, cert, ca, capath, cipher, cn) ((conn)->data)->m->ssl_set((conn)->data, (key), (cert), (ca), (capath), (cn), (cipher))
```
Comment (thetamind) [inline ext/mysqlnd/mysqlnd.h]: Transposing `cipher, cn` with `cn, cipher`intentional?

File: ext/mysqli/mysqli_api.c [hunk 4]
```diff
@@ -2258,7 +2258,7 @@ PHP_FUNCTION(mysqli_ssl_set)
 		}
 	}
 
-	mysql_ssl_set(mysql->mysql, ssl_parm[0], ssl_parm[1], ssl_parm[2], ssl_parm[3], ssl_parm[4]);
+	mysql_ssl_set(mysql->mysql, ssl_parm[0], ssl_parm[1], ssl_parm[2], ssl_parm[3], ssl_parm[4], NULL);
```
Comment (thetamind) [inline ext/mysqli/mysqli_api.c]: Intentionally not updating the `mysqli_ssl_set` api?

Comment (jerairrest) [inline ext/mysqli/mysqli_api.c]: mysqli_ssl_set definition now requires 6 parameters natively but my verison needs 7 to pass through the server's CA

**Pull request merged**